### ISSUE TITLE
DIM: Panic due to nil pointer

### ISSUE THREAD

**Issue by pranavgaikwad** [konveyor/mig-controller#922]
**Describe the bug**
When running direct image migration, I saw the controller panic due to a missing nil check: 
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1583f0b]

goroutine 1076 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /home/pranav/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20181127025237-2b1284ed4c93/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1f440a0, 0x37b5570)
        /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1.(*DirectImageMigration).getDISMProgress(0xc00086a400, 0xc000574138, 0x1, 0x1, 0x222857c, 0x7, 0x0, 0x1, 0xc00064d3b0)
        /home/pranav/go/src/github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1/directimagemigration_types.go:203 +0x12b
github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1.(*DirectImageMigration).HasCompleted(0xc00086a400, 0xc00086a400, 0x0, 0x0, 0x4, 0x4, 0xc001b8e380, 0xc000232000)
        /home/pranav/go/src/github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1/directimagemigration_types.go:192 +0x36a
github.com/konveyor/mig-controller/pkg/controller/migmigration.(*Task).Run(0xc001b99b28, 0x0, 0x0)
        /home/pranav/go/src/github.com/konveyor/mig-controller/pkg/controller/migmigration/task.go:440 +0x4cac
github.com/konveyor/mig-controller/pkg/controller/migmigration.(*ReconcileMigMigration).migrate(0xc000595890, 0xc001a42fc0, 0x2, 0x2, 0x0)
        /home/pranav/go/src/github.com/konveyor/mig-controller/pkg/controller/migmigration/migrate.go:65 +0x422
```

The associated DIM resource looks like this:

```
apiVersion: migration.openshift.io/v1alpha1
kind: DirectImageMigration
[...]
status:
  conditions:
  - category: Advisory
    lastTransitionTime: "2021-01-26T17:53:43Z"
    message: 'Step: 6/8'
    reason: CreateDirectImageStreamMigrations
    status: "True"
    type: Running
  - category: Required
    lastTransitionTime: "2021-01-26T17:53:02Z"
    message: Image migration is ready
    status: "True"
    type: Ready
  itinerary: PVC
  newISs:
  - destNamespace: bz-test
    name: centos
    namespace: bz-test
  observedDigest: fc90a0c2c83f7fc502d90653eb0e55b402c7122753153f4deef24ba88eb75f2c
  phase: CreateDirectImageStreamMigrations
  startTimestamp: "2021-01-26T17:53:02Z"
```

Since the imagestream reference in the `newISs` field above doesn't have a `DirectMigration` property, the controller panics here: https://github.com/konveyor/mig-controller/blob/0a9a5d8ef300d2091a56ea7ef4cb97c1b6a91711/pkg/apis/migration/v1alpha1/directimagemigration_types.go#L203

**To Reproduce**
1. In source cluster. Create a new namespace

  oc new-project bz-test

2. In source cluster. Create an internal imagestream in this namespace

  oc new-build -D $'FROM centos:7\nRUN yum install -y httpd' -n bz-test

3. Execute the migration for bz-test namespace

**Expected behavior**
Controller should not panic

**Screenshots & Snippets**

**Additional context**

**Comment by sseago** [comment]
Fix is here: https://github.com/konveyor/mig-controller/pull/978

**Comment by pranavgaikwad** [comment]
Fixed through #978
